pylsp/workspace.py

```diff
@@ -152,6 +153,9 @@ def _progress_begin(
         percentage: Optional[int] = None,
     ) -> str:
         token = str(uuid.uuid4())
+
+        self._endpoint.request(self.M_INITIALIZE_PROGRESS, {'token': token}).result(_utils.CALL_TIMEOUT)
```

Comment:
To quote the spec:

> To keep the protocol backwards compatible servers are only allowed to use `window/workDoneProgress/create` request if the client signals corresponding support using the client capability `window.workDoneProgress`
>
> https://microsoft.github.io/language-server-protocol/specifications/lsp/3.17/specification/#serverInitiatedProgress

And also it's a request so the server needs to wait for the result and is then only allowed to use that token if the result is successful.

Reply:
that's a valid point. I also checked rust-analyzer source, they also do it. 

I added checking for the client capability.